<!-- -->
<launch>

  <arg name="rviz" default="false" />
  <arg name="filesDirectory" default="$(env HOME)/lidar_slam/map" />
  <arg name="scanPeriod" default="0.1" />

  <node name="sensor_rot_TransPublish" pkg="tf" type="static_transform_publisher" args="0 0 0 -1.570796 -1.570796 0  lidar velodyne  30"/>

  <node name="lidar_init_TransPublish" pkg="tf" type="static_transform_publisher" args="0 0 0 1.570796  0 1.570796  lidar_init lidar_rot  30"/>

  <node pkg="lidar_slam" type="multi_scan_registration_node" name="multi_scan_registration_node" output="screen">
    <remap from="/multi_scan_points" to="/pandar_points" />
    <remap from="/imu/data" to="/imu_data"/>
    <param name="lidar" value="Pandar40" /> <!-- options: VLP-16  HDL-32  HDL-64E Pandar40 -->
    <param name="scanPeriod" value="$(arg scanPeriod)" />
    <param name="nFeatureRegions" value="6" />
    <param name="curvatureRegion" value="5" />
    <param name="maxCornerSharp" value="2" />
    <param name="maxCornerLessSharp" value="20" />
    <param name="maxSurfaceFlat" value="4" />
    <param name="surfaceCurvatureThreshold" value="0.05" />
    <param name="cornerCurvatureThreshold" value="1.0" />
    <param name="lessFlatFilterSize" value="0.2" />
    <param name="cornerCheckEnable" value="false" />

  </node>

  
  <!--node pkg="lidar_slam" type="laser_odometry_node" name="laser_odometry_node" output="screen" respawn="true"-->
  <node pkg="lidar_slam" type="laser_odometry_node" name="laser_odometry_node" output="screen" respawn="true">
  </node>

  <!--node pkg="lidar_slam" type="laser_mapping2_node" name="laser_mapping_node" output="screen"-->
  <node pkg="lidar_slam" type="laser_mapping_node" name="laser_mapping_node" output="screen">
  	<param name="filesDirectory" value="$(arg filesDirectory)"/>
  </node>

  <node pkg="lidar_slam" type="transform_maintenance_node" name="transform_maintenance_node" output="screen">
  </node>

  <node pkg="lidar_slam" type="graph_node" name="graph_node" output="screen">
        	<param name="filesDirectory" value="$(env HOME)/lidar_slam"/>
  </node>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find lidar_slam)/rviz_cfg/lidar_slam.rviz" />
  </group>
</launch>
